### PR TITLE
AF-1: Added a login form and user module

### PR DESCRIPTION
**Pull request by vlasichok**
AF-1: Added a login form using LoginComponent and UserModule, also added ProfileComponent for future development.

### CODE REVIEW

File: afisha-app/src/app/app.routes.ts [hunk 1]
```diff
@@ -8,5 +8,6 @@ export const appRoutes: Routes = [
   { path: 'events', component: EventsListComponent },
   { path: 'events/:id', component: EventComponent, canActivate: [EventRouteActivatorService] },
   { path: '404', component: Error404Component },
-  { path: '', redirectTo:'/events', pathMatch: 'full'}
+  { path: 'user', loadChildren: 'app/user/user.module#UserModule' },
```
Comment (doslechescake) [inline afisha-app/src/app/app.routes.ts]: Move all path strings to PATH const. 
Create /shared folder -> in this folder create /constant folder -> and create path.constant.ts file.
In this file create const PATH with all route paths and use this const instead string.

File: afisha-app/src/app/navbar/navbar.component.html [hunk 2]
```diff
@@ -4,7 +4,7 @@
   <button mat-button [matMenuTriggerFor]="animals">Members</button>
 
   <mat-menu #animals="matMenu">
-    <button mat-menu-item>Login</button>
+    <button mat-menu-item [routerLink] = "['/user/login']" >Login</button>
```
Comment (doslechescake) [inline afisha-app/src/app/navbar/navbar.component.html]: Use path property instead string.

File: afisha-app/src/app/user/login.component.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  loginForm: FormGroup;
+  private userEmail;
+  private userPassword;
+
+  constructor(private router: Router) {
+
+  }
+
+  ngOnInit() {
+    this.userEmail = new FormControl('',[Validators.required, Validators.email]);
```
Comment (doslechescake) [inline afisha-app/src/app/user/login.component.ts]: Import FormBuilder and use it

File: afisha-app/src/app/user/login.component.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    this.userPassword = new FormControl('',[Validators.required]);
+    this.loginForm = new FormGroup({
+      userEmail: this.userEmail,
+      userPassword: this.userPassword
+    });
+  }
+
+  cancel(){
+    this.router.navigate(['events']);
```
Comment (doslechescake) [inline afisha-app/src/app/user/login.component.ts]: Use PATH const.

File: afisha-app/src/app/user/user.module.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  imports: [
+    CommonModule,
+    RouterModule.forChild(userRoutes),
+    ReactiveFormsModule
+  ],
+  declarations: [
+    LoginComponent
+  ]
+})
+export class UserModule {
```
Comment (doslechescake) [inline afisha-app/src/app/user/user.module.ts]: What is UserModule? Maybe AuthModule will be more suit for this situations.

File: afisha-app/src/index.html [hunk 6]
```diff
@@ -9,8 +9,12 @@
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link rel="icon" type="image/x-icon" href="favicon.ico">
   <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css" integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm" crossorigin="anonymous">
```
Comment (doslechescake) [inline afisha-app/src/index.html]: Import scripts through angular-cli file

File: afisha-app/src/app/app.routes.ts [hunk 7]
```diff
@@ -3,10 +3,12 @@ import {Error404Component} from "./error/error404.component";
...
 import {EventRouteActivatorService} from "./events/services/event-route-activator.service";
+import {PATH} from "./shared/constants/path.constant";
 
 export const appRoutes: Routes = [
-  { path: 'events', component: EventsListComponent },
-  { path: 'events/:id', component: EventComponent, canActivate: [EventRouteActivatorService] },
-  { path: '404', component: Error404Component },
-  { path: '', redirectTo:'/events', pathMatch: 'full'}
+  { path: PATH.EVENTS, component: EventsListComponent },
+  { path: PATH.EVENTS_ID, component: EventComponent, canActivate: [EventRouteActivatorService] },
```
Comment (doslechescake) [inline afisha-app/src/app/app.routes.ts]: EventRouteActivatorService is a Guard. Please rename and move to appropriate folder.

**Comment by vlasichok** [comment]
@doslechescake Please check

File: afisha-client/src/app/user/login.component.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+      userEmail: ['', [Validators.required, Validators.email]],
+      userPassword: ['', [Validators.required]]
+    });
+  }
+
+  ngOnInit() {
+
+  }
+
+  cancel(){
```
Comment (doslechescake) [inline afisha-client/src/app/user/login.component.ts]: Please, write response type. In this case is 'void'

File: afisha-client/src/app/user/login.component.ts [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    this.router.navigate([PATH.EVENTS]);
+  }
+
+  login(loginValues) {
+    if(this.validateLoginForm()){
+      //call method to login the user
+    }
+    console.log("form is not valid");
```
Comment (doslechescake) [inline afisha-client/src/app/user/login.component.ts]: Write instead console.log TODO in this format -> // @TODO _text_

**Comment by vlasichok** [comment]
Please check, it's changed.

**Pull request closed without merging**